### PR TITLE
Add matomo events

### LINKED ISSUE #525: Matomo: Create New Events
# Feature Request

**As a** product manager
**I would** like more data on the usage of PeopleMover features
**so that** I can better understand what needs to be improved

## Acceptance Criteria

Category: Space Name
Action: Add Role

Category: Space Name
Action: Assign Role

Category: Space Name
Action: Edit Role

Category: Space Name
Action: Delete Role

Category: Space Name
Action: Edit Space Name

Category: Space Name
Action: Download Report

Category: Space Name
Action: Filter

Category: Space Name
Action: Sort

Category: Space Name
Action: Revert

Category: Space Name
Action: Archive Product

Category: Space Name
Action: Edit Person

Category: Space Name
Action: Add Location Tag

Category: Space Name
Action: Edit Location Tag

Category: Space Name
Action: Delete Location Tag

Category: Space Name
Action: Add Product Tag

Category: Space Name
Action: Edit Product Tag

Category: Space Name
Action: Delete Product Tag

## Notes

Refer to Sara for questions. Alex & Chris completed the last Matomo story.

### PR DESCRIPTION
**Pull request by alexwojtala**
## Issue
Resolves #525 
## What was done
- [x] Add more events to matomo

- [x] Category: Space Name Action: Add Role
- [ ] Category: Space Name Action: Assign Role - what would this be
- [x] Category: Space Name Action: Edit Role
- [x] Category: Space Name Action: Delete Role - gives id instead of name
- [x] Category: Space Name Action: Edit Space Name
- [x] Category: Space Name Action: Download Report
- [x] Category: Space Name Action: Filter
- [x] Category: Space Name Action: Sort
- [x] Category: Space Name Action: Revert
- [x] Category: Space Name Action: Archive Product
- [x] Category: Space Name Action: Edit Person
- [x] Category: Space Name Action: Add Location Tag
- [x] Category: Space Name Action: Edit Location Tag
- [x] Category: Space Name Action: Delete Location Tag
- [x] Category: Space Name Action: Add Product Tag
- [x] Category: Space Name Action: Edit Product Tag
- [x] Category: Space Name Action: Delete Product Tag

## How to test
1. Run through the actions outlined in the story
2. Ensure all these events are showing up in matomo

### CODE REVIEW

File: ui/src/Roles/RoleClient.ts [hunk 1]
```diff
@@ -16,10 +16,13 @@
...
 import Axios, {AxiosResponse} from 'axios';
-import {RoleAddRequest} from './RoleAddRequest';
-import {RoleEditRequest} from './RoleEditRequest';
 import {TraitClient} from '../Traits/TraitClient';
 import {getToken} from '../Auth/TokenProvider';
+import MatomoEvents from "../Matomo/MatomoEvents";
+import {Trait} from "../Traits/Trait";
+import {TraitAddRequest} from "../Traits/TraitAddRequest";
+import {Space} from "../Space/Space";
+import {TraitEditRequest} from "../Traits/TraitEditRequest";
```
Comment (RandomAshes) [inline ui/src/Roles/RoleClient.ts:25]: Please fix lint errors (double quotes)

Comment (RandomAshes) [reply ui/src/Roles/RoleClient.ts:25]: Also, the traits folder and files shouldn't exist anymore... why is there traits?

File: ui/src/Roles/RoleClient.ts [hunk 2]
```diff
@@ -38,40 +41,58 @@ class RoleClient implements TraitClient {
...
             headers: {
                 'Content-Type': 'application/json',
                 'Authorization': `Bearer ${getToken()}`,
             },
         };
 
-        return Axios.put(url, role, config);
+        return Axios.put(url, editRequest, config).then((result) => {
+            MatomoEvents.pushEvent(space.name, "editRole", editRequest.updatedName!!);
+            return result;
```
Comment (RandomAshes) [inline ui/src/Roles/RoleClient.ts:73]: I'm getting a 400 bad request when I try to edit a role..

**Review by RandomAshes — CHANGES_REQUESTED** [review]

File: ui/src/Roles/RoleClient.ts (hunk 2 above)
Comment (willot) [reply ui/src/Roles/RoleClient.ts:72]: You need to restart your back end some object got modified

File: ui/src/ReusableComponents/ProductFilter.tsx [hunk 3]
```diff
@@ -121,19 +122,12 @@ function ProductFilter({
...
-                    };
-                } else {
-                    return {
-                        ...option,
-                        selected: false,
-                    };
+            option => (
+                {
+                    ...option,
+                    selected: selectedOptions.includes(option),
```
Comment (aclift19Ford) [inline ui/src/ReusableComponents/ProductFilter.tsx:128]: I'm seeing this break:
Screen Shot 2020-11-16 at 1 48 16 PM

**Comment by alexwojtala** [comment]
[x] Category: Space Name Action: Add Role

[ ] Category: Space Name Action: Assign Role - what would this be

[x] Category: Space Name Action: Edit Role

[x] Category: Space Name Action: Delete Role - gives id instead of name

[x] Category: Space Name Action: Edit Space Name

[x] Category: Space Name Action: Download Report

[x] Category: Space Name Action: Filter

[x] Category: Space Name Action: Sort

[x] Category: Space Name Action: Revert

[x] Category: Space Name Action: Archive Product

[x] Category: Space Name Action: Edit Person

[x] Category: Space Name Action: Add Location Tag

[x] Category: Space Name Action: Edit Location Tag

[x] Category: Space Name Action: Delete Location Tag

[x] Category: Space Name Action: Add Product Tag

[x] Category: Space Name Action: Edit Product Tag

[x] Category: Space Name Action: Delete Product Tag

**Review by aclift19Ford — CHANGES_REQUESTED** [review]

**Comment by willot** [comment]
We can't easily know if the role of a person is changed. We know is a person is updated but not excatly what field is updated.

**Review by michaelrygiel — APPROVED** [review]
⭐ ⭐ ⭐ ⭐ ⭐

**Review by aclift19Ford — APPROVED** [review]
Looks good

**Pull request merged**